extensions/packageManager/src/client/feedModal.tsx

```diff
@@ -0,0 +1,178 @@
+      setEditRow(true);
+    } else {
+      setEditRow(false);
+    }
+  }, [selectedItem?.key]);
+
+  const columns = [
+    { key: 'column1', name: 'Name', fieldName: 'text', minWidth: 100, maxWidth: 250, isResizable: true, onRender: (item: PackageSourceFeed) => {
+      if (!selectedItem || item.key !== selectedItem.key || !editRow) return item.text;
+      return <Fragment>
+          <TextField
+            placeholder={formatMessage('Feed Name')}
+            value={selectedItem ? selectedItem.text : ''}
+            disabled={!selectedItem || selectedItem.readonly}
+            onChange={updateSelected('text')}
+          />
+      </Fragment> },
+    },
+    { key: 'column2', name: 'URL', fieldName: 'url', minWidth: 450, isResizable: true, onRender: (item: PackageSourceFeed) => {
+      if (!selectedItem || item.key !== selectedItem.key || !editRow) return item.url;
+      return <Fragment>
+          <TextField
+            placeholder={formatMessage('URL')}
+            value={selectedItem ? selectedItem.url : ''}
+            disabled={!selectedItem || selectedItem.readonly}
+            onChange={updateSelected('url')}
+          />
+      </Fragment> },
+    },
+    { key: 'column3', minWidth: 80, isResizable: false, name: '', onRender: (item: PackageSourceFeed) => {
+      if (selectedItem && item.key === selectedItem.key && !editRow)
+      return <Fragment>
+        <IconButton iconProps={{ iconName: 'Edit' }} onClick={()=>setEditRow(true)} disabled={!selectedItem || selectedItem.readonly} />
+        <IconButton iconProps={{ iconName: 'Delete' }} onClick={removeSelected} disabled={!selectedItem || selectedItem.readonly} />
+      </Fragment>;
+
+      if (selectedItem && item.key === selectedItem.key && editRow)
+      return <Fragment>
+        <IconButton iconProps={{iconName: 'Checkmark'}} onClick={() => { setEditRow(false); props.onUpdateFeed(selectedItem.key, selectedItem) }}
+                    disabled={!selectedItem || !selectedItem.text || !selectedItem.url || selectedItem.readonly}
+         />
+      </Fragment>;
+    },}
+  ];
+
+  const updateSelected = (field: string) => {
+    return (evt, val) => {
+      const newSelection = {
+        ...selectedItem,
+        [field]: val,
+      };
+      setSelectedItem(newSelection);
+    };
+  };
+
+  const addItem = () => {
+    const newItem = {
+      key: uuid(),
+      text: 'New Feed',
+      url: 'http://',
+    } as PackageSourceFeed;
+
+    const newItems = items.concat([newItem]);
+
+    // select newest to display in the editor
+    setSelectedItem(newItem);
+
+    // update the detail view
+    setItems(newItems);
+
+    // update the selection
+    selection.setItems(newItems, true);
+    selection.setKeySelected(newItem.key, true, false);
+    setSelection(selection);
+  };
+
+  const removeSelected = async () => {
+    if (await confirm(formatMessage('Delete this feed?'), formatMessage('Are you sure you want to remove this feed source?'))) {
```

Comment:
Can these texts be localized?

Reply:
not sure what you mean here.  isn't formatMessage the way to localize them?